src/test/globalHooks.ts

```diff
@@ -0,0 +1,30 @@
+// Global Mocha test hooks.
+
+import * as util from 'util';
+
+const original = {
+	log: console.log,
+	error: console.error,
+};
+
+beforeEach(function () {
+	const currentTest = this.currentTest as {
+		consoleOutputs?: string[],
+		consoleErrors?: string[],
+	};
+	console.log = function captureLog() {
+		original.log.apply(console, arguments);
+		const formatted = util.format.apply(util, arguments);
+		currentTest.consoleOutputs = (currentTest.consoleOutputs || []).concat(formatted);
+	};
+	console.error = function captureError() {
+		original.error.apply(console, arguments);
+		const formatted = util.format.apply(util, arguments);
+		currentTest.consoleErrors = (currentTest.consoleErrors || []).concat(formatted);
+	};
```

Comment:
This _does_ result in stdout and stderr sections being placed in the JUnit XML output, but it doesn't look like Azure DevOps is exposing it for anything. It shouldn't hurt though.